android/src/main/java/com/swmansion/rnscreens/utils/ScreenDummyLayoutHelper.kt

```diff
@@ -57,25 +58,47 @@ internal class ScreenDummyLayoutHelper(
     }
 
     /**
-     * Initializes dummy view hierarchy with CoordinatorLayout, AppBarLayout and dummy View.
+     * Tries to initialize dummy view hierarchy with CoordinatorLayout, AppBarLayout and dummy View.
      * We utilize this to compute header height (app bar layout height) from C++ layer when its needed.
      *
+     * This method might fail in case there is activity attached to the react context.
+     *
+     * This method is called from various threads!
+     *
      * @return boolean whether the layout was initialised or not
      */
     private fun maybeInitDummyLayoutWithHeader(reactContext: ReactApplicationContext): Boolean {
         if (isLayoutInitialized) {
             return true
         }
 
+        // Possible data race here - activity is injected into context on UI thread.
         if (!reactContext.hasCurrentActivity()) {
             return false
         }
 
         // We need to use activity here, as react context does not have theme attributes required by
         // AppBarLayout attached leading to crash.
         val contextWithTheme =
-            requireNotNull(reactContext.currentActivity) { "[RNScreens] Attempt to use context detached from activity" }
+            requireNotNull(reactContext.currentActivity) { "[RNScreens] Attempt to use context detached from activity. This could happen only due to race-condition." }
```

Comment:
Why aren't you using `reactContextRef` everywhere?

Reply:
This method takes `reactContext: ReactApplicationContext` as a parameter, there is no point of using the `reactContextRef` field here. It makes more sense for me to take this as parameter, because:

1. this method requires nonnull react context to do anything,
2. moving context validation out of the method allows to throw better error messages which are tailored to specific code path & context, see e.g. `onHostResume` method;

Reply:
I do not see any advantages of using `reactContextRef` directly, because I do need to sanitise the reference anyway an almost all cases, because I do want to throw / log an message everytime `reactContextRef.get() == null` & it is unexpected.